src/schneider_lab_to_nwb/zempolich_2024/zempolich_2024_convert_session.py

```diff
@@ -0,0 +1,194 @@
+    # conversion_options.update(dict(ISOI=dict()))
+
+    converter = Zempolich2024NWBConverter(source_data=source_data, verbose=verbose)
+    metadata = converter.get_metadata()
+
+    # Update default metadata with the editable in the corresponding yaml file
+    editable_metadata_path = Path(__file__).parent / "zempolich_2024_metadata.yaml"
+    editable_metadata = load_dict_from_file(editable_metadata_path)
+    metadata = dict_deep_update(metadata, editable_metadata)
+
+    add_session_start_time_to_metadata(
+        behavior_file_path=behavior_file_path, ephys_folder_path=ephys_folder_path, metadata=metadata
+    )
+
+    if has_ephys:
+        conversion_options["Sorting"]["units_description"] = metadata["Sorting"]["units_description"]
+
+    # # Overwrite video metadata
+    # for i, video_file_path in enumerate(video_file_paths):
+    #     metadata_key_name = f"VideoCamera{i+1}"
+    #     metadata["Behavior"][metadata_key_name] = editable_metadata["Behavior"][metadata_key_name]
+
+    subject_id = behavior_file_path.name.split("_")[1]
+    session_id = behavior_file_path.name.split("_")[2]
+    nwbfile_path = output_dir_path / f"sub-{subject_id}_ses-{session_id}.nwb"
+    metadata["NWBFile"]["session_id"] = session_id
+    metadata["Subject"]["subject_id"] = subject_id
+
+    # Run conversion
+    converter.run_conversion(metadata=metadata, nwbfile_path=nwbfile_path, conversion_options=conversion_options)
+
+
+def add_session_start_time_to_metadata(
+    behavior_file_path: str | Path,
+    ephys_folder_path: Optional[str | Path],
+    metadata: dict,
+):
+    if ephys_folder_path is not None:
+        folder_name = ephys_folder_path.parent.name + "/" + ephys_folder_path.name
+        folder_name_to_start_datetime = metadata["Ecephys"].pop("folder_name_to_start_datetime")
+        if folder_name in folder_name_to_start_datetime.keys():
+            session_start_time = datetime.fromisoformat(folder_name_to_start_datetime[folder_name])
+        else:
+            session_start_time = metadata["NWBFile"]["session_start_time"]
+    else:
+        session_start_time = datetime.strptime(behavior_file_path.name.split("_")[2], "%y%m%d")
+
+    EST = ZoneInfo("US/Eastern")
+    metadata["NWBFile"]["session_start_time"] = session_start_time.replace(tzinfo=EST)
+
+
+def main():
+    # Parameters for conversion
+    data_dir_path = Path("/Volumes/T7/CatalystNeuro/Schneider/Grant Zempolich Project Data")
+    output_dir_path = Path("/Volumes/T7/CatalystNeuro/Schneider/conversion_nwb")
+    stub_test = False
+    verbose = False
+
+    if output_dir_path.exists():
+        shutil.rmtree(output_dir_path, ignore_errors=True)
+
+    # Example Session A1 Ephys + Behavior
+    ephys_folder_path = data_dir_path / "A1_EphysFiles" / "m53" / "Day1_A1"
+    behavior_file_path = data_dir_path / "A1_EphysBehavioralFiles" / "raw_m53_231029_001.mat"
+    video_folder_path = Path("")
```

Comment:
empty path?

Reply:
Yes, video and intrinsic signal imaging haven't yet been provided for the full dataset, but they should be coming soon after thanksgiving.